packages/core/test/app/ovm/quantifiers/merkle-inclusion-quantifier.spec.ts

```diff
@@ -0,0 +1,172 @@
+        result.results.length === 1,
+        'There should only be one result that is undefined.'
+      )
+      assert(
+        result.results[0] === undefined,
+        'There should only be one result that is undefined.'
+      )
+      assert(
+        !result.allResultsQuantified,
+        'All results should not be quantified if we are unable to find a leaf node in the tree.'
+      )
+    })
+
+    it('returns undefined on populated tree with a root that does not match', async () => {
+      const quantifier = new MerkleInclusionQuantifier(merkleTree)
+
+      merkleTree.setLeaf(key, Buffer.from('VALUE'))
+
+      const result: QuantifierResult = await quantifier.getAllQuantified({
+        key,
+        root: Buffer.from('some root'),
+      })
+
+      assert(
+        result.results.length === 1,
+        'There should only be one result that is undefined.'
+      )
+      assert(
+        result.results[0] === undefined,
+        'There should only be one result that is undefined.'
+      )
+      assert(
+        !result.allResultsQuantified,
+        'All results should not be quantified if we are unable to find a leaf node in the tree.'
+      )
+    })
+
+    it('returns undefined on empty tree with a root that matches', async () => {
+      const quantifier = new MerkleInclusionQuantifier(merkleTree)
+      const root: Buffer = Buffer.from('some root')
+
+      merkleTree.setRootHash(root)
+
+      const result: QuantifierResult = await quantifier.getAllQuantified({
+        key,
+        root,
+      })
+
+      assert(
+        result.results.length === 1,
+        'There should only be one result that is undefined.'
+      )
+      assert(
+        result.results[0] === undefined,
+        'There should only be one result that is undefined.'
+      )
+      assert(
+        !result.allResultsQuantified,
+        'All results should not be quantified if we are unable to find a leaf node in the tree.'
+      )
+    })
+
+    it('returns value on populated tree tree with a root that matches', async () => {
+      const quantifier = new MerkleInclusionQuantifier(merkleTree)
+      const root: Buffer = Buffer.from('some root')
+      const value: Buffer = Buffer.from('VALUE')
+
+      merkleTree.setRootHash(root)
+      merkleTree.setLeaf(key, value)
+
+      const result: QuantifierResult = await quantifier.getAllQuantified({
+        key,
+        root,
+      })
+
+      assert(
+        result.results.length === 1,
+        'There should only be one result that is the value.'
+      )
+      assert(
+        result.results[0].equals(value),
+        `Result value [${result.results[0].toString()}] found when expecting [${value.toString()}]`
+      )
+      assert(
+        result.allResultsQuantified,
+        'All results should be quantified if we are able to find the leaf node in the tree.'
+      )
+    })
+
+    it('returns value on populated tree tree with a root that matches but mismatched key', async () => {
```

Comment:
"returns value"--should this be "returns undefined" or am I missing something?

Reply:
Yes, thanks for reading thoroughly 😄